src/application-manager.coffee

```diff
@@ -0,0 +1,875 @@
+			_.find(pendingPairs, (pair) -> pair.target?.serviceName == dependency)? or _.find(stepsInProgress, (step) -> step.target?.serviceName == dependency)?
+		return false if dependencyUnmet
+		# for networks and volumes, check no network pairs have that volume name
+		if _.find(networkPairs, (pair) -> pair.target.name == target.network_mode)?
+			return false
+		if _.find(stepsInProgress, (step) -> step.model == 'network' and step.target.name == target.network_mode)?
+			return false
+		volumeUnmet = _.some target.volumes, (volumeDefinition) ->
+			sourceName = volumeDefinition.split(':')[0]
+			_.find(volumePairs, (pair) -> pair.target.name == sourceName)? or _.find(stepsInProgress, (step) -> step.model == 'volume' and step.target.name == sourceName)?
+		return !volumeUnmet
+
+	# Unless the update strategy requires an early kill (i.e. kill-then-download, delete-then-download), we only want
+	# to kill a service once the images for the services it depends on have been downloaded, so as to minimize
+	# downtime (but not block the killing too much, potentially causing a deadlock)
+	_dependenciesMetForServiceKill: (target, targetApp, availableImages) ->
+		if target.depends_on?
+			for dependency in target.depends_on
+				dependencyService = _.find(targetApp.services, (s) -> s.serviceName == dependency)
+				if !_.find(availableImages, (image) -> image.name == dependencyService.image)?
+					return false
+		return true
+
+	_nextStepsForNetworkOrVolume: ({ current, target }, currentApp, changingPairs, dependencyComparisonFn, model) ->
+		# Check none of the currentApp.services use this network or volume
+		if current?
+			dependencies = _.filter currentApp.services, (service) ->
+				dependencyComparisonFn(service, current)
+			if _.isEmpty(dependencies)
+				return [{ action: 'removeNetworkOrVolume', model, current }]
+			else
+				# If the current update doesn't require killing the services that use this network/volume,
+				# we have to kill them before removing the network/volume (e.g. when we're only updating the network config)
+				steps = []
+				_.forEach dependencies, (dependency) ->
+					if !_.some(changingPairs, (pair) -> pair.serviceId == dependency.serviceId)
+						steps.push(serviceAction('kill', dependency.serviceId, dependency))
+				return steps
+		else if target?
+			return [{ action: 'createNetworkOrVolume', model, target }]
+
+	_nextStepsForNetwork: ({ current, target }, currentApp, changingPairs) =>
+		dependencyComparisonFn = (service, current) ->
+			service.network_mode == current.name
+		@_nextStepsForNetworkOrVolume({ current, target }, currentApp, changingPairs, dependencyComparisonFn, 'network')
+
+	_nextStepsForVolume: ({ current, target }, currentApp, changingPairs) ->
+		# Check none of the currentApp.services use this network or volume
+		dependencyComparisonFn = (service, current) ->
+			_.some service.volumes, (volumeDefinition) ->
+				sourceName = volumeDefinition.split(':')[0]
```

Comment:
This will cause a normal volume to be incorectly interpreted as a named volume

Reply:
Fixed (though it wouldn't have really caused problems, since the name would never have matched a named volume's name)